system/jlib/jsuperhash.cpp

```diff
@@ -448,6 +448,13 @@ bool SuperHashTable::removeExact(void *et)
     return true;
 }
 
+unsigned SuperHashTable::validIdx(unsigned i) const
+{
+    while (i < tablesize && !table[i])
+        i++;
+    return i;
```

Comment:
@ghalliday - What happens if there is no valid i ? e.g. no elements or beyond last? 
I didn't spot any error checking on begin() calling firstIdx() ..

Reply:
the maximum i can be is tablesize.  That should be the value of hash.end() - so it will not iterate through anything.